src/merge.js

```diff
@@ -243,8 +243,9 @@ class Merge {
         doc.creationDate = file.creationDate
         if (this.sameBinary(file, doc)) {
           if (doc.size == null) { doc.size = file.size }
-          if (doc.class == null) { doc.class = file.class }
-          if (doc.mime == null) { doc.mime = file.mime }
+
+          doc.class = file.class
+          doc.mime = file.mime
```

Comment:
Actually the looping stops as soon as this fix is enabled.
But you'right it won't work: it will actually prevent changing the mimetype once the file's metadata have been created in pouch.
I'll need to come up with another solution.